File: project/5.EX_総合理解度チェック/java/java_ex5_ex_1/src/java_ex5_ex_1/Ex5_ex_1.java
package java_ex5_ex_1;

class Rock {
	byte weight;
	
	Rock( byte w ){
		if( 10 >= w ) {
			this.weight = w;
		}
		else {
			this.weight = 10;
		}
	}
}

public class Ex5_ex_1 {

	public static void main(String[] args) {
		// オブジェクトの配列を確保して、for文内で個別にインスタンス生成
		Rock[] rc = new Rock[10];
		for( int num = 0; num < 10; num++ ) {
			rc[num] = new Rock( (byte)(num+1) ); 
		}
		
		int sum = 0;
		for( int idx = 0; idx < rc.length; idx++ ) {
			sum += rc[idx].weight;
			System.out.println( rc[idx].weight );
		}
		
		System.out.println( sum );
		
	}

}

File: project/5.EX_総合理解度チェック/py/py_ex5_ex_1/src/ex5_ex_1.py
'''
Created on 2023/05/05

@author: Keroichi.T
'''

class Rock( object ):
    def __init__(self, w):
        if w <= 10 :
            self.weight = w
        else:
            self.weight = 10

def main():
    rcs = []
    for num in range(10):
        rcs.append( Rock(num+1) ) 

    tmp = 0
    for rc in rcs:
        tmp += rc.weight
        print( rc.weight )
    
    print( tmp )


if __name__ == '__main__':
    main()